asmcomp/arm64/emit.mlp

```diff
@@ -643,8 +657,14 @@ let emit_instr i =
           match addr with
           | Iindexed _ -> i.arg.(1)
           | Ibased(s, ofs) ->
-              `	adrp	{emit_reg reg_tmp1}, {emit_symbol_offset s ofs}\n`;
-              reg_tmp1 in
+              if not !Clflags.dlcode then begin
+                `	adrp    {emit_reg reg_tmp1}, {emit_symbol_offset s ofs}\n`
+              end else begin
+                `	adrp	{emit_reg reg_tmp1}, :got:{emit_symbol s}\n`;
+                `	ldr	{emit_reg reg_tmp1}, [{emit_reg reg_tmp1}, #:got_lo12:{emit_symbol s}]\n`
+              end;
+              reg_tmp1
+	in
```

Comment:
Likewise.  Perhaps this code could be factored between `Iload` and `Istore`, but we'll see later.